src/test/java/org/example/service/AuthServiceTests.java

```diff
@@ -0,0 +1,73 @@
+package org.example.service;
+
+import io.jsonwebtoken.Jwts;
+import org.example.daos.AuthDao;
+import org.example.exceptions.InvalidException;
+import org.example.models.LoginRequest;
+import org.example.models.User;
+import org.example.services.AuthService;
+import org.example.validators.LoginValidator;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.sql.SQLException;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+
+class AuthServiceTests {
+
+    AuthDao authDao = Mockito.mock(AuthDao.class);
+    LoginValidator loginValidator = spy(new LoginValidator());
+    Key jwtKey = Jwts.SIG.HS256.key().build();
+    AuthService authService= new AuthService(authDao, loginValidator, jwtKey);
+
+    LoginRequest loginRequest = new LoginRequest(
+            "admin@kainos.com",
+            "wlSNgEn5dCBM59jnbeH+txKWn36Vt6QScELcAa5ZBNduqSY16JAl2hqeGsZrmpG0kdb9+ILMoCJVB3er8ZoCJI9o26IM83UfnJtTT3p7cRgOUxsU0iMHgkI9KdQpDim6"
+    );
+
+    LoginRequest validLoginRequest = new LoginRequest(
+            "admin@kainos.com",
+            "dFupuh0ITKJyNJEEyHSWwHRd5Mnhm65I+WfTUZWbOh8I+akEWJ6VDsFsWetFCdA1EQP+Mx193tVd7MudxjAMoBYqusV78vypAzv/TDfKYVmZCVsco8q7hskFtsWnKVCf"
+
+    );
+
+    User user = new User(
+            "admin@kainos.com",
+            "wlSNgEn5dCBM59jnbeH+txKWn36Vt6QScELcAa5ZBNduqSY16JAl2hqeGsZrmpG0kdb9+ILMoCJVB3er8ZoCJI9o26IM83UfnJtTT3p7cRgOUxsU0iMHgkI9KdQpDim6",
+            "salt",
```

Comment:
Missing test cases when the Dao returns null (one for each Dao method)

Reply:
This has been corrected, tested, and committed.